src/plugins/visualizations/public/vis.ts

```diff
@@ -137,19 +149,25 @@ export class Vis<TVisParams = VisParams> {
     if (state.data && (state.data.aggs || !this.data.aggs)) {
       const aggs = state.data.aggs ? cloneDeep(state.data.aggs) : [];
       const configStates = this.initializeDefaultsFromSchemas(aggs, this.type.schemas.all || []);
-      if (!this.data.indexPattern) {
-        if (aggs.length) {
-          const errorMessage = i18n.translate(
-            'visualizations.initializeWithoutIndexPatternErrorMessage',
-            {
-              defaultMessage: 'Trying to initialize aggs without index pattern',
-            }
-          );
-          throw new Error(errorMessage);
-        }
-        return;
+
+      if (!this.data.indexPattern && aggs.length) {
+        this.data.indexPattern = new DataView({
+          spec: {
+            id: state.data.searchSource?.index,
+            type: 'fallback',
```

Comment:
I'm pretty sure this is not what `type` is meant for - it's to capture rollup vs regular data views.. Also, isn't the `title` the actual index pattern string?

Reply:
What about using the default data view instead if we need to fill in something?

Reply:
ok. updated that  to more legal way